platform/admin/configurator/admin-configurator-impl/src/main/java/org/codice/ddf/admin/configurator/impl/OperationReportImpl.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.admin.configurator.impl;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import org.codice.ddf.admin.configurator.OperationReport;
+import org.codice.ddf.admin.configurator.Result;
+
+public class OperationReportImpl implements OperationReport {
+    private final Map<String, Result> results = new LinkedHashMap<>();
+
+    public boolean isTransactionSucceeded() {
+        return results.values()
+                .stream()
+                .allMatch(Result::isTransactionSucceeded);
+    }
+
+    public Result getResult(String key) {
+        return results.get(key);
+    }
+
+    public List<Result> getFailedResults() {
+        return Collections.unmodifiableList(results.values()
```

Comment:
I would be fine here not wrapping it as modifiable. My previous comment was just documenting at the top that the returned list would not impact internals of the report implementation and since you are collecting them in a new list, there won't be any possibilities to mess up the internals of this class with a non-modifable list returned this way.